cockatrice/src/filter_string.cpp

```diff
@@ -0,0 +1,217 @@
+#include "filter_string.h"
+#include "../../common/lib/peglib.h"
+
+#include <QByteArray>
+#include <QString>
+#include <cmath>
+#include <functional>
+
+peg::parser search(R"(
```

Comment:
This long block of parser code just worries me... Does this change the editor's efficiency?

Reply:
Nah it gets compiled to an intermediate form on app start.